### PR TITLE
Add support for server password

### LINKED ISSUE #172: Is there support for IRC server password?
I have a password set up to connect my IRC server. Is there an option to configure this in the .env file?

### PR DESCRIPTION
**Pull request by Tjzabel**
This PR enables TeleIRC to connect to IRC servers that require an authenticated password.

Closes #172

### CODE REVIEW

**Comment by Tjzabel** [comment]
@HolmesSherlock do these changes cover what you did for the server password authentication?

I don't have access to an IRC server that uses password auth, and so there's no viable way for me to test this functionality.

**Comment by HolmesSherlock** [comment]
@Tjzabel Commit #186 works for me, I confirm.

**Comment by Tjzabel** [comment]
@HolmesSherlock thanks for the quick reply!

File: docs/config-file-glossary.rst [hunk 1]
```diff
@@ -35,6 +35,9 @@ IRC settings
 ``IRC_SERVER=chat.freenode.net``
     IRC server to connect to
 
+``IRC_SERVER_PASSWORD=""``
+    IRC server password
+
```
Comment (jwflory) [inline docs/config-file-glossary.rst]: Thanks for keeping the docs up-to-date! :100:

**Comment by jwflory** [comment]
For the record, this also landed in v1.3.3.

**Pull request merged**